arch/x86/core/fatal.c

```diff
@@ -471,29 +469,29 @@ void z_x86_page_fault_handler(z_arch_esf_t *esf)
 	dump_page_fault(esf);
 #endif
 #ifdef CONFIG_THREAD_STACK_INFO
-	if (z_x86_check_stack_bounds(esf_get_sp(esf), 0, esf->cs)) {
-		z_x86_fatal_error(K_ERR_STACK_CHK_FAIL, esf);
+	if (z_x86_check_stack_bounds(esf_get_sp(esf), 0, (uint16_t)esf->cs)) {
+		z_x86_fatal_error((unsigned int)K_ERR_STACK_CHK_FAIL, esf);
```

Comment:
What's up here?  If the defined error enumerants don't work to pass to the fatal error handler, we have an API bug.  A cast is the wrong solution.